src/dotnet/Fable.Compiler/Transforms/Replacements.fs

```diff
@@ -1401,6 +1406,498 @@ let unchecked (com: ICompiler) r t (i: CallInfo) (_: Expr option) (args: Expr li
     | "Compare" -> Helper.CoreCall("Util", "compare", t, args, i.ArgTypes, ?loc=r) |> Some
     | _ -> None
 
+
+let enums (_: ICompiler) r _ (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
+    match thisArg, i.CompiledName, args with
+    | Some this, "HasFlag", [arg] ->
+        // x.HasFlags(y) => (int x) &&& (int y) <> 0
+        makeBinOp r (Fable.Number Int32) this arg BinaryAndBitwise
+        |> fun bitwise -> makeEqOp r bitwise (makeIntConst 0)BinaryUnequal
+        |> Some
+    | _ -> None
+
+let log (com: ICompiler) r t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
+    let v =
+        match args with
+        | [] -> []
+        | [v] -> [v]
+        | (Fable.Value (Fable.StringConstant _))::_ ->
+            Helper.CoreCall("String", "format", t, args, i.ArgTypes)
+            |> List.singleton
+        | _ -> [args.Head]
+    Helper.GlobalCall("console", t, args, ?argTypes=None, ?thisArg=None, memb="log", ?loc=r)
```

Comment:
It's not necessary here to write `?argTypes=None`. Optional arguments only need the name when they don't appear in order (as `memb="log"` here) or when we pass an option value, prefixed with `?` (as in `?loc=r`), but we can omit them if the value is `None`.